### PR TITLE
{Keyvault} Fix keyvault packaging tests

### PR DESCRIPTION
**Pull request by evelyn-ys**
**Description**

Our CI fails in ([CI link](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1001522&view=logs&jobId=1e6fba43-bdcf-51f0-a062-3712c33fbb51&j=1e6fba43-bdcf-51f0-a062-3712c33fbb51&t=a1e787eb-ec67-5c3d-79ec-0c68f4f59d65))
image
with such error
```
Can't overwrite existing cassette ('/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/keyvault/tests/latest/recordings/test_keyvault_certificate_crud.yaml') in your current record mode ('once').
No match for the request (<Request (GET) https://cli-test-kv-ct-000002.vault.azure.net/secrets/cert2/?api-version=7.0>) was found.
```
when runs into L1895
https://github.com/Azure/azure-cli/blob/562eeaa45ddce309d3ff30723595ce3820631af5/src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py#L1892-L1895

It's because the `cert_secret_path` is `TEST_DIR/cert_secret` without any file extension and our packaging script doesn't cover file without file extension: https://github.com/Azure/azure-cli/blob/562eeaa45ddce309d3ff30723595ce3820631af5/scripts/ci/build.sh#L135-L168
So the code runs into this `if not os.path.exists(cert_secret_path)` but no related request found in yaml file.

This PR changed `TEST_DIR/cert_secret` to `TEST_DIR/cert_secret.der` and added `.der` to packaging file, also re-record yaml file with related request.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py [hunk 1]
```diff
@@ -1972,7 +1972,7 @@ def test_keyvault_certificate_crud(self, resource_group, key_vault):
         self.cmd('keyvault certificate delete --vault-name {kv} -n cert1')
         time.sleep(60)
         self.cmd('keyvault certificate purge --vault-name {kv} -n cert1')
-        time.sleep(10)
+        time.sleep(120)
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py:1975]: Is this change related to this PR?

**Comment by yonzhan** [comment]
Keyvault

File: scripts/ci/build.sh [hunk 2]
```diff
@@ -141,6 +141,7 @@ cat >>$testsrc_dir/setup.py <<EOL
                        '*.zip',
                        '*.pem',
                        '*.pfx',
+                       '*.der',
```
Comment (jiasli) [inline scripts/ci/build.sh:144]: If the issue is caused by a missing certificate file, why isn't the error triggered before?

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py (hunk 1 above)
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py:1975]:
```suggestion
        time.sleep(10)
```

File: scripts/ci/build.sh (hunk 2 above)
Comment (jiasli) [reply scripts/ci/build.sh:144]: I think `*.der` should be added to L157 as well.

File: src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py (hunk 1 above)
Comment (evelyn-ys) [reply src/azure-cli/azure/cli/command_modules/keyvault/tests/latest/test_keyvault_commands.py]: No it's for debug, I have reverted this change.

**Comment by jiasli** [comment]
You may directly push the branch to `azure` repo to trigger those CI tasks.

**Comment by jiasli** [comment]
Please include a direct link to fail CI tasks in the PR description, instead of using screenshots.

File: scripts/ci/build.sh (hunk 2 above)
Comment (evelyn-ys) [reply scripts/ci/build.sh:144]: done

Comment (jiasli) [reply scripts/ci/build.sh:144]: If the issue is caused by a missing certificate file, why isn't the error triggered before?

**Comment by jiasli** [comment]
If the issue is caused by a missing certificate file, why isn't the error triggered before?

**Comment by evelyn-ys** [comment]
> You may directly push the branch to `azure` repo to trigger those CI tasks.

Got it! Redirect to #18922
